File: app/src/main/java/com/synapse/social/studioasinc/widget/ZoomImageViewLib/animation/AnimCompat.java
package com.synapse.social.studioasinc.widget.ZoomImageViewLib.animation;

import android.os.Build;
import android.view.View;

public class AnimCompat {
    private static final int SIXTY_FPS_INTERVAL = 16;

    public static void postOnAnimation(View view, Runnable runnable) {
        if (Build.VERSION.SDK_INT >= SIXTY_FPS_INTERVAL) {
            view.postOnAnimation(runnable);
        } else {
            view.postDelayed(runnable, 16);
        }
    }
}

File: app/src/main/java/com/synapse/social/studioasinc/widget/ZoomImageViewLib/animation/AnimCompat.kt
package com.synapse.social.studioasinc.widget.ZoomImageViewLib.animation

import android.os.Build
import android.view.View

object AnimCompat {
    private const val SIXTY_FPS_INTERVAL = 16L

    fun postOnAnimation(view: View, runnable: Runnable) {
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.JELLY_BEAN) {
            view.postOnAnimation(runnable)
        } else {
            view.postDelayed(runnable, SIXTY_FPS_INTERVAL)
        }
    }
}
